src/python/deepgnn/graph_engine/snark/tests/e2e_test.py

```diff
@@ -24,87 +24,103 @@
 import deepgnn.graph_engine.snark.convert as convert
 import deepgnn.graph_engine.snark.dispatcher as dispatcher
 import deepgnn.graph_engine.snark._lib as lib
+from deepgnn.graph_engine.snark.converter.writers import BinaryWriter
+
+
+nodes = [
+    {
+        "node_id": 9,
+        "node_type": 0,
+        "node_weight": 1,
+        "uint64_feature": {"2": [13, 17]},
+        "float_feature": {"0": [0, 1], "1": [-0.01, -0.02]},
+        "binary_feature": {},
+        "edge": [],
+    },
+    {
+        "node_id": 0,
+        "node_type": 1,
+        "node_weight": 1,
+        "uint64_feature": {},
+        "float_feature": {"0": [1], "1": [-0.03, -0.04]},
+        "binary_feature": {"3": "abcd"},
+        "edge": [],
+    },
+    {
+        "node_id": 5,
+        "node_type": 2,
+        "node_weight": 1,
+        "float_feature": {"0": [1, 1], "1": [-0.05, -0.06]},
+        "binary_feature": {},
+        "uint8_feature": {"4": [5, 6, 7]},
+        "int8_feature": {"5": [15, 16, 17]},
+        "uint16_feature": {"6": [25, 26, 27]},
+        "int16_feature": {"7": [35, 36, 37]},
+        "uint32_feature": {"8": [45, 46, 47]},
+        "int32_feature": {"9": [55, 56, 57]},
+        "uint64_feature": {"10": [65, 66, 67]},
+        "int64_feature": {"11": [75, 76, 77]},
+        "double_feature": {"12": [85, 86, 87]},
+        "float16_feature": {"13": [95, 96, 97]},
+        "sparse_float16_feature": {
+            "14": {"coordinates": [5, 13], "values": [1.0, 2.13]}
+        },
+        "edge": [],
+    },
+]
+edges = [
+    {
+        "src_id": 9,
+        "dst_id": 0,
+        "edge_type": 0,
+        "weight": 0.5,
+        "uint64_feature": {"0": [1, 2, 3]},
+        "float_feature": {},
+        "binary_feature": {},
+    },
+    {
+        "src_id": 0,
+        "dst_id": 5,
+        "edge_type": 1,
+        "weight": 1,
+        "uint64_feature": {},
+        "float_feature": {"1": [3, 4]},
+        "binary_feature": {},
+    },
+    {
+        "src_id": 5,
+        "dst_id": 9,
+        "edge_type": 1,
+        "weight": 0.7,
+        "float_feature": {},
+        "binary_feature": {"2": "hello"},
+        "uint8_feature": {"4": [5, 6, 7]},
+        "int8_feature": {"5": [15, 16, 17]},
+        "uint16_feature": {"6": [25, 26, 27]},
+        "int16_feature": {"7": [35, 36, 37]},
+        "uint32_feature": {"8": [45, 46, 47]},
+        "int32_feature": {"9": [55, 56, 57]},
+        "uint64_feature": {"10": [65, 66, 67]},
+        "int64_feature": {"11": [75, 76, 77]},
+        "double_feature": {"12": [85, 86, 87]},
+        "float16_feature": {"13": [95, 96, 97]},
+        "sparse_float16_feature": {
```

Comment:
We test sparse features in a separate suite. Could you update `sparse_features_test.py`?